java/client/src/main/java/glide/api/models/commands/stream/StreamRange.java

```diff
@@ -0,0 +1,96 @@
+ * stream search by stream ID.
+ *
+ * @see <a href="https://redis.io/commands/xrange/">redis.io</a>
+ * @see <a href="https://redis.io/commands/xrevrange/">redis.io</a>
+ */
+public interface StreamRange {
+
+    String getRedisApi();
+
+    String MINIMUM_RANGE_REDIS_API = "-";
+    String MAXIMUM_RANGE_REDIS_API = "+";
+    String RANGE_COUNT_REDIS_API = "COUNT";
+
+    /**
+     * Enumeration representing minimum or maximum stream entry bounds for the range search, to get
+     * the first or last stream ID.
+     */
+    @RequiredArgsConstructor
+    @Getter
+    enum InfRangeBound implements StreamRange {
+        MIN(MINIMUM_RANGE_REDIS_API),
+        MAX(MAXIMUM_RANGE_REDIS_API);
+
+        private final String redisApi;
+    };
+
+    /**
+     * Stream ID used to specify a range of IDs to search. Stream ID bounds can be complete with a
+     * timestamp and sequence number separated by a dash (<code>"-"</code>), for example <code>
+     * "1526985054069-0"</code>.<br>
+     * Stream ID bounds can also be incomplete, with just a timestamp.<br>
+     * Stream ID bounds are inclusive by default. When <code>isInclusive==false</code>, a <code>"("
+     * </code> is prepended for the Redis API.
+     */
+    @Getter
+    class IdBound implements StreamRange {
+        private final String redisApi;
+
+        /**
+         * Creates an incomplete stream ID boundary without the sequence number for range search.
+         *
+         * @param timestamp The stream timestamp as ID.
+         */
+        public IdBound(long timestamp) {
+            redisApi = Long.toString(timestamp);
+        }
+
+        /**
+         * Creates an incomplete stream ID boundary without the sequence number for range search.
+         *
+         * @param timestamp The stream timestamp as ID.
+         * @param isInclusive Whether the ID range bound is inclusive. Defaults to true if not set.
+         */
+        public IdBound(long timestamp, boolean isInclusive) {
+            redisApi = isInclusive ? Long.toString(timestamp) : "(" + timestamp;
+        }
+
+        /**
+         * Creates a stream ID boundary by stream id for range search.
+         *
+         * @param id The stream id.
+         */
+        public IdBound(String id) {
+            redisApi = id;
+        }
+
+        /**
+         * Creates a stream ID boundary by stream id for range search.
+         *
+         * @param id The stream id.
+         * @param isInclusive Whether the ID range bound is inclusive. Defaults to true if not set.
+         */
+        public IdBound(String id, boolean isInclusive) {
+            redisApi = isInclusive ? id : "(" + id;
+        }
+    }
+
+    static String[] toArgs(StreamRange start, StreamRange end) {
+        return new String[] {start.getRedisApi(), end.getRedisApi()};
+    }
+
+    static String[] toArgs(StreamRange start, StreamRange end, long count) {
```

Comment:
Add some docs, because these methods exposed to a user, even though it should not.